mlflow/java/scoring/src/main/java/org/mlflow/models/Model.java

```diff
@@ -64,6 +65,11 @@ public static Model fromRootPath(String modelRootPath) throws IOException {
   public static Model fromConfigPath(String configPath) throws IOException {
     File configFile = new File(configPath);
     Model model = SerializationUtils.parseYamlFromFile(configFile, Model.class);
+    // Set the model uuid if it's absent.
+    if (!model.getModelUuid().isPresent()) {
+      String uuid = UUID.randomUUID().toString().replace("-", "");
+      model.setModelUuid(uuid);
```

Comment:
Shall we set a new ID under this case ? I prefer to keep it empty because the ID should be generated while logging model

Reply:
If we set new generated ID here, suppose we load a old version model twice, we will get 2 in-memory model with different model ID, this break the rule of model ID being immutable

Reply:
> If we set new generated ID here, suppose we load a old version model twice, we will get 2 in-memory model with different model ID, this break the rule of model ID being immutable

```python
import tempfile
import os

from mlflow.models import Model

with tempfile.TemporaryDirectory() as tmp_dir:
    path = os.path.join(tmp_dir, "MLmodel")

    with open(path, "w") as f:
        f.write(
            """
artifact_path: model
flavors:
  python_function:
    env: conda.yaml
    loader_module: mlflow.sklearn
    model_path: model.pkl
    python_version: 3.7.9
  sklearn:
    pickled_model: model.pkl
    serialization_format: cloudpickle
    sklearn_version: 0.24.1
"""
        )

    print(Model.load(path).model_uuid)
    print(Model.load(path).model_uuid)
```

output:

```
cac277097e54410c8d191a717e640cf6
560075c0351f43dca8554f8ab6eb96f4
```

Reply:
Yes. I create a fixing PR for python side:
https://github.com/mlflow/mlflow/pull/5167